liferay-workspace/tests/blade.samples.test/src/testFunctional/java/com/liferay/blade/samples/update/portlet/test/BladeSamplesUpdatePortletTest.java

```diff
@@ -0,0 +1,265 @@
+	}
+
+	@Test
+	public void testUpdateMVCPortletProject() throws Exception {
+		_webDriver.get(_portletURL.toExternalForm());
+
+		Assert.assertTrue(
+			"Portlet was not deployed", isVisible(_helloWorldPortlet));
+		Assert.assertTrue(
+			_portletTitle.getText(),
+			_portletTitle.getText().contentEquals("helloworld Portlet"));
+		Assert.assertTrue(
+			_portletBody.getText(),
+			_portletBody.getText().contentEquals("Hello from helloworld JSP!"));
+
+		File dynamicFile = new File(
+			_projectPath +
+			"/src/main/java/helloworld/portlet/HelloworldPortlet.java");
+
+		List<String> lines = new ArrayList<>();
+		String line = null;
+
+		try (BufferedReader reader =
+		new BufferedReader(new FileReader(dynamicFile))) {
+
+			while ((line = reader.readLine()) != null) {
+				lines.add(line);
+
+				if (line.equals("import javax.portlet.Portlet;")) {
+					String s =
+						new StringBuilder().
+						append("import java.io.IOException;\n").
+						append("import javax.portlet.PortletException;\n").
+						append("import javax.portlet.RenderRequest;\n").
+						append("import javax.portlet.RenderResponse;\n").
+						toString();
+
+					lines.add(s);
+				}
+
+				if (line.equals(
+						"public class HelloworldPortlet extends MVCPortlet {"))
+					{
+
+					String s =
+						new StringBuilder().append("public void doView(\n").
+							append("RenderRequest renderRequest," +
+								" RenderResponse renderResponse)\n").
+							append("throws IOException, PortletException {\n").
+							append("renderRequest.setAttribute(\n").
+							append("\"foo\", \"bar\");\n").
+							append("super.doView(renderRequest, " +
+								"renderResponse);\n").append("}\n").toString();
+
+					lines.add(s);
+				}
+			}
+		}
+
+		try (Writer writer = new FileWriter(dynamicFile)) {
+			for (String string : lines) {
+				writer.write(string + "\n");
+			}
+		}
+
+		File staticFile = new File(
+			_projectPath +
+			"/src/main/resources/META-INF/resources/view.jsp");
+
+		lines = new ArrayList<>();
+		line = null;
+
+		try (BufferedReader reader =
+		new BufferedReader(new FileReader(staticFile))) {
+
+			while ((line = reader.readLine()) != null) {
+				lines.add(line);
+
+				if (line.contains("liferay-ui:message key=")) {
+					String s =
+						new StringBuilder().
+						append("<b><%= renderRequest.getAttribute(\"foo\") %></b>").
+						toString();
+
+					lines.add(s);
+				}
+			}
+		}
+
+		try (Writer writer = new FileWriter(staticFile)) {
+			for (String string : lines) {
+				writer.write(string + "\n");
+			}
+		}
+
+		try {
+			_buildStatus = BladeCLIUtil.execute(_projectPath, "gw", "assemble");
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+
+		Assert.assertTrue(
+			"Expected Build Successful, but saw: " + _buildStatus,
+			!_buildStatus.contains("failed"));
+
+		File buildOutput = new File(
+			_projectPath + "/build/libs/helloworld-1.0.0.jar");
+
+		new JMXBundleDeployer().deploy(_helloWorldJarBSN, buildOutput);
+
+		Thread.sleep(10000);
```

Comment:
Do we need this?  Once deploy returns the bundle should be ready to go, I think we can just hit the form directly?